packages/sdk/src/modules/liquidation/getPoolUsers.ts

```diff
@@ -43,20 +44,26 @@ async function getFusePoolUsers(
   const poolUsers: PoolUserStruct[] = [];
   const comptrollerInstance = sdk.createComptroller(comptroller);
   const borrowersCount = await comptrollerInstance.callStatic.getAllBorrowersCount();
+  console.log(`Total borrowers count for ${comptroller}: ${borrowersCount.toNumber()}`); // Print borrower count for the current pool
+
   const randomPage = Math.round(Math.random() * borrowersCount.div(PAGE_SIZE).toNumber());
   const [_totalPages, users] = await comptrollerInstance.callStatic.getPaginatedBorrowers(randomPage, PAGE_SIZE);
-  for (const user of users) {
-    const assets = await sdk.contracts.PoolLens.callStatic.getPoolAssetsWithData(comptrollerInstance.address, {
-      from: user
-    });
 
+  const assetsPromises = users.map((user) =>
+    sdk.contracts.PoolLens.callStatic.getPoolAssetsWithData(comptrollerInstance.address, { from: user })
+  );
+
+  const assetsResults = await Promise.all(assetsPromises);
+
+  assetsResults.forEach((assets, index) => {
     const { totalBorrow, totalCollateral } = getUserTotals(assets);
     const health = getPositionHealth(totalBorrow, totalCollateral);
 
     if (maxHealth.gt(health)) {
-      poolUsers.push({ account: user, totalBorrow, totalCollateral, health });
+      poolUsers.push({ account: users[index], totalBorrow, totalCollateral, health });
```

Comment:
We can't do this because we are pushing to the same array in parallel. This will cause concurrency issues. Need to handle this in the promise resolution after promise.all is done.

Reply:
sure let me make change accordingly

Reply:
made some changes please review